pyof/foundation/network_types.py

```diff
@@ -3,14 +3,48 @@
 Defines and Implements Basic Network packet types , such as Ethertnet and LLDP.
 """
 
+# System imports
+from enum import IntEnum
+
+# Local source tree imports
 from pyof.foundation.base import GenericStruct
 from pyof.foundation.basic_types import (
     BinaryData, HWAddress, IPAddress, UBInt8, UBInt16)
-from pyof.foundation.constants import VLAN_TPID
 from pyof.foundation.exceptions import PackException, UnpackException
 
-__all__ = ('ARP', 'Ethernet', 'GenericTLV', 'IPv4', 'VLAN', 'TLVWithSubType',
-           'LLDP')
+__all__ = ('ARP', 'Ethernet', 'EtherType', 'GenericTLV', 'IPv4', 'VLAN',
+           'TLVWithSubType', 'LLDP')
+
+
+# NETWORK CONSTANTS AND ENUMS
+class EtherType(IntEnum):
+    """Enumeration with IEEE Ethernet types.
+
+    The items are being added as we need.
+    If you need one EtherType that is not listed below, please, send us a Pull
+    Request with the addition.
+
+    Ref: http://standards-oui.ieee.org/ethertype/eth.txt
+
+    """
+
+    #: Internet Protocol version 4 (IPv4)
+    IPV4 = 0x0800
+    #: Address Resolution Protocol (ARP)
+    ARP = 0x0806
+    #: Reverse Address Resolution Protocol
+    RARP = 0x8035
+    #: VLAN-tagged frame (IEEE 802.1Q) and Shortest Path Bridging IEEE 802.1aq
+    #: with NNI compatibility[8]
+    VLAN = 0x8100
+    #: Internet Protocol Version 6 (IPv6)
+    IPV6 = 0x86DD
+    #: Ethernet flow control
+    ETHERNET_FLOW_CONTROL = 0x8808
+    #: Link Layer Discovery Protocol (LLDP)
+    LLDP = 0x88CC
+    #: VLAN-tagged (IEEE 802.1Q) frame with double tagging
+    VLAN_DOUBLE_TAG = 0x9100
```

Comment:
Maybe the name VLAN_QINQ would make sense here?
Besides that, 0x9100 is an outdated value for an old protocol - the value for QinQ S-TAGs is 0x88a8

Reply:
Thanks for the suggestion, VLAN_QINQ added (and value fixed)